packages/core/rfcs/setting-composable-internal-fields.md

```diff
@@ -0,0 +1,55 @@
+# Setting composable internal fields
+
+## Motivation
+
+Sometimes when we create a new composable (using factory) we need to override some fields that are inside.
+For example, when you use `useProduct` you want to reload `cart` property from the `useCart` composable.
+To do this our factories can return setter function instead of just composable instance:
+
+### Factory:
+```js
+const useCartFactory = () => {
+  const cart = ref({});
+
+  const setCart = (newCart) => {
+    cart.value = newCart;
+  }
+
+  const useCart = () => {
+    return { cart }
+  };
+
+  return { setCart, useCart };
+}
+
+export default useCartFactory
+```
+
+### Integration:
+```js
+import { useCartFactory } from '@vue-storefront/factories';
+
+export default useCartFactory({ ... })
+```
+
+now usage (example with user):
+
+```js
+import { setCart } from './useCart';
```

Comment:
would need to be like this
```suggestion
import { useCart } from 'composablesPackage';

const {setCart} = useCart()
```

Reply:
no, that would be an internal api from factory, Andrzej was right